EXADGConfig.cmake.in

```diff
@@ -0,0 +1,15 @@
+set(EXADG_VERSION @VERSION@)
+
+@PACKAGE_INIT@
+
+get_filename_component(EXADG_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
```

Comment:
Where is EXADG_CMAKE_DIR used?

i do not really understand what EXADGConfig.cmake.in is doing? Maybe you could explain this a bit.

Reply:
EXADGConfig.cmake.in serves as a template. Running CMake will create EXADGConfig.cmake and EXADGTargets.cmake from it in the build folder. These files are needed to use the FIND_PACKAGE command in other projects. 

EXADG_CMAKE_DIR as well as EXADG_INCLUDE_DIRS are available after the FIND_PACKAGE call and can be used in an external project. See e.g. the description of the pull request, where EXADG_INCLUDE_DIRS is used to make the include directories available in the external project. The more we export, the easier it will be to interface.  With EXADG_CMAKE_DIR the user has the build directory of EXADG at his fingertips, even though it might not be needed by everybody.

Similar to that, I also export EXADG_VERSION which is currently not set since EXADG currently has no version number. In the future this might be helpful though.

Reply:
For me this looks like great work! To get the present PR completed, I would like you to include parts of the above answer as comments in the code.

In CMakeLists.txt, please describe this:

> EXADGConfig.cmake.in serves as a template. Running CMake will create EXADGConfig.cmake and EXADGTargets.cmake from it in the build folder. These files are needed to use the FIND_PACKAGE command in other projects.

At the end of CMakeLists.txt, you should write the three lines in the description of this PR how to use ExaDG in other projects. This is very helpful for us because not everyone is familiar with CMake.

> find_package(EXADG REQUIRED)
target_link_libraries(<target_which_links_to_exadg> EXADG::exadg)
target_include_directories(<target_which_links_to_exadg> ${EXADG_INCLUDE_DIRS})

In the EXADGConfig.cmake.in file, you should describe the variables that are exported:

> EXADG_CMAKE_DIR as well as EXADG_INCLUDE_DIRS are available after the FIND_PACKAGE call and can be used in an external project. Similarly, EXADG_VERSION is also exported (which might be helpful in the future once EXADG has a version number).

As a minor comment, could you remove multiple empty lines?

Reply:
Done :)